### PR TITLE
fix issue 15545, csv Reader line feed '\r' failure

### PR DESCRIPTION
**Pull request by BBasile**

### CODE REVIEW

File: std/csv.d [hunk 1]
```diff
@@ -1016,8 +1016,11 @@ public:
            if (_input.range.front == '\r')
            {
                _input.range.popFront();
-               if (_input.range.front == '\n')
-                   _input.range.popFront();
+               if (!_input.range.empty)
+               {
+                    if (_input.range.front == '\n')
```
Comment (burner) [inline std/csv.d]: you can join the two if statements

File: std/csv.d [hunk 2]
```diff
@@ -1127,6 +1130,19 @@ public:
     }
 }
 
+// Bugzilla 15545
+unittest
```
Comment (JackStouffer) [inline std/csv.d]: `@safe pure unittest`

Comment (BBasile) [inline std/csv.d]: the unittest cannot be safe because its _try catch_ but pure ok

**Comment by BBasile** [comment]
#3917 updated and squashed.

File: std/csv.d [hunk 3]
```diff
@@ -1013,14 +1013,14 @@ public:
...
         {
-           if (_input.range.front == '\r')
-           {
-               _input.range.popFront();
-               if (_input.range.front == '\n')
-                   _input.range.popFront();
-           }
-           else if (_input.range.front == '\n')
-               _input.range.popFront();
+            if (_input.range.front == '\r')
```
Comment (burner) [inline std/csv.d]: you added some unwanted leading space

Comment (BBasile) [inline std/csv.d]: No identation is alright:

https://github.com/BBasile/phobos/blob/82d6dbf8175055e1851bd6601dc20d0f5649e2f7/std/csv.d#L1014

Comment (burner) [inline std/csv.d]: sry my bad, I didn't see that the old indent was missing one space

**Comment by burner** [comment]
LGTM and you where first to create you PR

**Comment by burner** [comment]
Auto-merge toggled on

**Pull request merged**